lib/launcher.js

```diff
@@ -360,24 +355,51 @@ class Launcher {
             this.argv[frameworkOptsKey][frameworkTimeoutKey] = 10 * 60 * 1000
         }
 
-        this.processes.push(childProcess)
-
-        childProcess
-            .on('message', this.messageHandler.bind(this, cid))
-            .on('exit', this.endHandler.bind(this, cid))
-
-        childProcess.send({
-            cid,
-            command: 'run',
-            configFile: this.configFile,
-            argv: this.argv,
-            caps,
-            processNumber,
-            specs,
-            server,
-            isMultiremote: this.isMultiremote()
-        })
+        let childProcess
+        let retries = this.configParser.getConfig().specFileRetries || 0
+        let spawn = () => {
+            let messages = []
+            let idx = this.processes.indexOf(childProcess)
+
+            childProcess = child.fork(path.join(__dirname, '/runner.js'), process.argv.slice(2), {
+                cwd: process.cwd(),
+                execArgv
+            })
+
+            this.processes.splice(idx === -1 ? this.processes.length : idx, 1, childProcess)
+
+            childProcess.on('message', message => {
+                message._timestamp = new Date()
+                if (retries === 0) {
+                    this.messageHandler(cid, message)
```

Comment:
Just got an idea from the Gitter channel, let's add the retry count to the message object so we can propagate the info to the reporters.

Reply:
The way things currently work will only ever send messages to the reporters when `retries === 0`.
When `retries > 0` and the specfile fails, buffered messages are discarded, and the reporters would be unaware that the failing testrun ever existed in the first place.

The reason for this is so that existing code continues to work. I suspect that if we suddenly report failing tests to reporters (even if they are retried soon afterwards) then a lot of things in the field will probably break.

Reply:
> then a lot of things in the field will probably break.

Good point, let's leave stuff as is then

Reply:
I'm all for the idea of reporting the failures and retries to reporters, as this would allow for more insight into what happened. I just suspect things would become slightly more involved then than they currently are.
Mocha's suite-level and test-level retries are transparent (to my knowledge). Perhaps the specfile-level retries should be too.
Your call, ultimately.